design-proposals/volume-claim-templates.md

```diff
@@ -0,0 +1,117 @@
+# Overview
+This proposal is about having a better integration with kubernetes storage APIs.  
+With recent support of kubernetes [volume populators](https://kubernetes.io/blog/2022/05/16/volume-populators-beta/) we have implemented [CDI volume populators](https://github.com/kubevirt/containerized-data-importer/blob/main/doc/cdi-populators.md) which allows population of PVC the kubernetes way without the need for DataVolumes.  
+Along major benefits of using this populators we loss one of DataVolumes greatest achievements - generating a PVC with the optimal properties fitting the user's storage.  
+This design should help us gain the best of all worlds.
+
+## Motivation
+Harnessing k8s and CDI new volume populators API along with our knowledge of storage capabilities will allow the user to enjoy synchronization with k8s and better compatibility with backup and restore solutions as well as the smart creation of PVCs that best suits the user's storage.
+
+## Goals
+- Provide a method for creating VMs with PVCs that suits best the user's storage while using CDI volume populators.
+
+## Non Goals
+
+## User Stories
+* As a user creating VMs, I want to be able to create a PVC within the VM spec while getting the best PVC spec that is also compatible with backup and restore solutions.
+
+## Repos
+* **CDI**: Provide API to that receives a PVC spec and in case of missing properties such as VolumeMode and AccessModes fills the optimal values according to the storage class.
+* **CDI**: Update DataImportCron API to create/update VolumeCloneSources that can be consumed as sources for PVC handled by CDI volume populators. VolumeCloneSource will need to be able to be updated with different PVC source.
+* **Kubevirt**: Provide API to create a VM with minimal volume information and receive a VM with best suited populated PVC.
+* **Kubevirt**: Provide virtctl command to generate and create PVCs.
+* **SSP**: Manage setting up VolumeCloneSources for commonly supported operating systems updated by DataImportCrons.
+* **Common Templates**: Transition to use VolumeClaimTemplates in VM specs and use CDI's VolumeCloneSource as the source in the template.
+* **OCP UI**: Provide API to create a CDI Volume populator CR that can be used by the volumeClaim. In case of golden image source need to ensure VolumeCloneSource is linked properly to a PVC in order to determine whether the boot image for a common template is available.
```

Comment:
I'm not sure this is an API but rather just an internal implementation detail.  In any case, anything that creates VMs (UI, virtctl, tekton pipelines, etc) may want to adjust to using the new kubevirt volumeClaimTemplates API.  It should be noted that this can be done incrementally since we do not propose to eliminate the dataVolumeTemplates section abruptly.  We may want to consider deprecating it at some point in the future though.

Reply:
the available boot image part I saw a similar thing mentioned in the golden images doc so I added it here. I can remove it no problem.
I'll mention here what is the first planned steps and whats later on. I did mention it in the last part of the implementation phases

Reply:
I don't think the design of the OCP UI belongs in a KubeVirt design proposal FWIW.

Reply:
ill remove the common templates and UI parts and mention that any repo creating a vm should take this new API into account and prefer to use it as default instead of datavolumetemplate